### PR TITLE
Updates profile validation

### PR DESCRIPTION
**Pull request by mohitb35**
Resolves potential errors faced due to mismatch in server/client validation for the profile form fields

Changes:

1. Updates validation rules in complete signup/edit profile form fields
2. Reorganizes translations, grouping text for field labels and validation errors together, and removing unreqd translation files from `serverSideTranslations`. This is done for all languages (except for new text which is in english only)
3. Adds dependencies
    a. babel/plugin-transform-unicode-regex - to allow unicode regexp patterns as used in server validation

### CODE REVIEW

**Review by mariahosfeld — COMMENTED** [review]
Reviewed functionality within edit profile, worked as expected there.

**Comment by mohitb35** [comment]
> [Plant-for-the-Planet-org/planet-donations#446 (comment)](https://github.com/Plant-for-the-Planet-org/planet-donations/pull/446#issuecomment-1805665718) This applies here as well. Apart from this, everything seems fine to me.

Resolved in [8b0b916](https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1920/commits/8b0b916ea5c2a84c59cd9698356d8a2181d93788)

**Comment by prachigarg19** [comment]
> > [Plant-for-the-Planet-org/planet-donations#446 (comment)](https://github.com/Plant-for-the-Planet-org/planet-donations/pull/446#issuecomment-1805665718) This applies here as well. Apart from this, everything seems fine to me.
> 
> Resolved in [8b0b916](https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1920/commits/8b0b916ea5c2a84c59cd9698356d8a2181d93788)

I'm still able to add the allowed special characters for the respective field at the start of input
image

**Comment by mariahosfeld** [comment]
> > > [Plant-for-the-Planet-org/planet-donations#446 (comment)](https://github.com/Plant-for-the-Planet-org/planet-donations/pull/446#issuecomment-1805665718) This applies here as well. Apart from this, everything seems fine to me.
> > 
> > 
> > Resolved in [8b0b916](https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1920/commits/8b0b916ea5c2a84c59cd9698356d8a2181d93788)
> 
> I'm still able to add the allowed special characters for the respective field at the start of input image

In general these are the special characters that are allowed for usage. But I agree, we should make sure that while they can be used within the fields, the first character in the field cannot be a special character.

**Review by mariahosfeld — COMMENTED** [review]
Functionality looks good to me.

**Review by prachigarg19 — APPROVED** [review]

File: public/static/locales/en/editProfile.json [hunk 1]
```diff
@@ -32,5 +22,39 @@
...
+    "nameInvalid": "Name is invalid. Only these special characters are allowed: space . , ' & ( ) ! -",
+    "addressRequired": "Address is required",
+    "addressInvalid": "Address is invalid. Only these special characters are allowed: space . , # - /",
+    "cityRequired": "City is required",
+    "cityInvalid": "City is invalid. Only letters and these special characters are allowed: space . , ( ) -",
+    "zipCodeRequired": "Zip Code is required",
+    "zipCodeInvalid": "Zip Code is invalid",
+    "websiteInvalid": "Please enter valid Website URL",
+    "countryRequired": "Country is required",
+    "companyRequired": "Company Name is required"
```
Comment (norbertschuler) [inline public/static/locales/en/editProfile.json:58]: This resource does not seem to be used (I assume nameRequired is used instead)?

**Pull request merged**